### ISSUE TITLE
`spy` does not retain the function result object if it is a promise

### ISSUE THREAD

**Issue by flunderpero** [tinylibs/tinyspy#21]
I have the following vitest test:

```
vi.mock(f).mockImplementation(() => {
   const res = new Promise()
   res.cancel = () => ....
   return res
})
```

When `f` is then called inside the production code, `f.cancel` is no longer there because - as far as I can see in the source code - tinyspy _replaces_ the promise returned from `mockImplementation` [with its result](https://github.com/tinylibs/tinyspy/blob/cb19dd1642042dea16197fc4dfc7a57586f15660/src/spy.ts#L69).

Edit: This is working in Jest, btw. (I am in the process of migration our Jest test code base to vitest)

Edit 2: If I remove the special handling of promises, all our tests run just fine. I did not dig deep enough into the code to fully understand why that code is there in the first place, so please forgive me for making dumb suggestions.

**Comment by sheremet-va** [comment]
You are correct in your assumptions. This is an oversight, you can open a PR to fix this, if you have time. Otherwise I will fix it tomorrow

**Comment by sheremet-va** [comment]
Fixed in v0.3.3
